initialize_package.sh

```diff
@@ -0,0 +1,360 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# Constants and help message.
+OLD_NAME="template_component_package"
+HELP_MESSAGE="Usage: $0 [--dry-run]
+This script initializes the package by prompting you to choose which component variants to include:
+  - Python Lifecycle Component
+  - Python Component
+  - C++ Lifecycle Component
+  - C++ Component
+
+Based on your choices, it will set up (or remove) the corresponding files and directories.
+
+Options:
+  -n, --dry-run    Display the actions without making any changes.
+  -h, --help       Show this help message."
+
+DRY_RUN=false
+
+# Process options.
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -n|--dry-run)
+      DRY_RUN=true
+      shift
+      ;;
+    -h|--help)
+      echo "${HELP_MESSAGE}"
+      exit 0
+      ;;
+    -*)
+      echo "Unknown option: $1" >&2
+      echo "${HELP_MESSAGE}"
+      exit 1
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+# Logging helper.
+log() {
+  echo "$1"
+}
+
+# Prompt until nonempty input is given.
+prompt_nonempty() {
+  local prompt_msg="$1"
+  local input
+  while true; do
+    read -p $'\e[3;36m'"${prompt_msg}"$'\e[0m ' input
+    if [[ -n "${input}" ]]; then
+      echo "${input}"
+      return
+    else
+      log "Input cannot be empty. Please try again."
+    fi
+  done
+}
+
+# Prompt for yes/no input with a default of "no" if empty.
+prompt_yesno() {
+  local prompt_msg="$1"
+  local ans
+  while true; do
+    read -p $'\e[3;36m'"${prompt_msg} [y/n]:"$'\e[0m ' ans
+    # If no input, default to "n"
+    if [[ -z "$ans" ]]; then
+      ans="n"
```

Comment:
```suggestion
      ans="y"
```

For me the default should be yes to also reflect the current state

Reply:
or we simply merge that as a standalone option into the same script (probably the best)